### ISSUE TITLE
How filter result by reference column?

### ISSUE THREAD

**Issue by fu-hsi** [mbdavid/LiteDB#1625]
Hi.
This is not work:

`var result = db.GetCollection<Product>("products").Include(x => x.Category).Find(x => x.Category.Id == 14).ToList();`

I receive empty result. 
This work:

`var result = db.GetCollection<Product>("products").Include(x => x.Category).FindAll().Where(x => x.Category.Id == 14).ToList();`

What I am doing wrong? How to filter products by category?
```
[BsonRef("categories")]
public Category Category { get; set; }
```

**Comment by lbnascimento** [comment]
@fu-hsi Your first example should be working. Could you provide us the datafile you're using?

**Comment by fu-hsi** [comment]
Hi.
Now I use Repository-Pattern because of this, but I will try again and give you more details soon.

**Comment by fu-hsi** [comment]
```
    public class Category
    {
        [BsonId]
        public int Id { get; set; }
        public string Name { get; set; }
    }
```

```
    public class Product
    {
        [BsonId]
        public int Id { get; set; }

        [BsonRef("categories")]
        public Category Category { get; set; }

        public string Name { get; set; }
    }
```

Data in database:
```
[
  {
    "Id": 1,
    "Category": {
      "Id": 1,
      "Name": null
    },
    "Name": "Produkt 12.04.2020 08:57:26",
  }
]
```
Not works (with Include):
```
var result = db.GetCollection<Product>("products").Include(x => x.Category).Find(x=>x.Category.Id == 1).ToList();
```

Works (without Include):
```
var result = db.GetCollection<Product>("products").Include(x => x.Category).Find(x=>x.Category.Id == 1).ToList();
```

**Comment by lbnascimento** [comment]
@fu-hsi The `Category` data was inserted incorrectly: it was inserted as a subdocument of `Product` instead of a reference to another collection. You should remove this entry, insert the categories in a separate collection (if you haven't already) and then insert the product data again (the existence of `BsonRefAttribute` should be enough for it to be inserted correctly this time).

**Comment by fu-hsi** [comment]
This is only data dump from result variable.
Insert query:

```
products.Insert(new Product { Name = "Produkt " + DateTime.Now, Category = category });
```
Data from LiteDB Studio:

```
/* 1 */
{
  "_id": 1,
  "Category": 
  {
    "$id": 1,
    "$ref": "categories"
  },
  "Name": "Produkt 12.04.2020 08:57:26",
}
```

**Comment by lbnascimento** [comment]
@fu-hsi This issue has been fixed and its fix is already available in version 5.0.7.

**Comment by fu-hsi** [comment]
Yes, it is OK now, thanks!
